jmx-scraper/src/test/java/io/opentelemetry/contrib/jmxscraper/config/JmxScraperConfigTest.java

```diff
@@ -138,74 +142,96 @@ void shouldFailValidation_missingConfigPathAndTargetSystem() {
-  //    JmxConfig config = new JmxConfig(props);
-  //
-  //    // This property should retain the system property value, not the config file value
-  //    assertThat(config.serviceUrl).isEqualTo("myServiceUrl");
-  //    // These properties are set from the config file
-  //    assertThat(config.groovyScript).isEqualTo("/my/groovy/script");
-  //    assertThat(config.targetSystem).isEqualTo("jvm,cassandra");
-  //    assertThat(config.targetSystems).containsOnly("jvm", "cassandra");
-  //    assertThat(config.intervalMilliseconds).isEqualTo(20000);
-  //    assertThat(config.metricsExporterType).isEqualTo("otlp");
-  //    assertThat(config.otlpExporterEndpoint).isEqualTo("https://myotlpendpoint");
-  //    assertThat(config.prometheusExporterHost).isEqualTo("host123.domain.com");
-  //    assertThat(config.prometheusExporterPort).isEqualTo(67890);
-  //    assertThat(config.username).isEqualTo("myUser\nname");
-  //    assertThat(config.password).isEqualTo("myPassw\\ord");
-  //    assertThat(config.remoteProfile).isEqualTo("SASL/DIGEST-MD5");
-  //    assertThat(config.realm).isEqualTo("myRealm");
-  //
-  //    // This property should retain the system property value, not the config file value
-  //    assertThat(System.getProperty("javax.net.ssl.keyStorePassword")).isEqualTo("truth");
-  //    // These properties are set from the config file loading into JmxConfig
-  //    assertThat(System.getProperty("javax.net.ssl.keyStore")).isEqualTo("/my/key/store");
-  //    assertThat(System.getProperty("javax.net.ssl.keyStoreType")).isEqualTo("JKS");
-  //    assertThat(System.getProperty("javax.net.ssl.trustStore")).isEqualTo("/my/trust/store");
-  //    assertThat(System.getProperty("javax.net.ssl.trustStorePassword")).isEqualTo("def456");
-  //    assertThat(System.getProperty("javax.net.ssl.trustStoreType")).isEqualTo("JKS");
-  //  }
-  //
-  //  @Test
-  //  @SetSystemProperty(key = "otel.jmx.service.url", value = "myServiceUrl")
-  //  @SetSystemProperty(key = "otel.jmx.groovy.script", value = "myGroovyScript")
-  //  @SetSystemProperty(key = "otel.jmx.target.system", value = "myTargetSystem")
-  //  void canSupportScriptAndTargetSystem() {
-  //    JmxConfig config = new JmxConfig();
-  //
-  //    assertThat(config.serviceUrl).isEqualTo("myServiceUrl");
-  //    assertThat(config.groovyScript).isEqualTo("myGroovyScript");
-  //    assertThat(config.targetSystem).isEqualTo("mytargetsystem");
-  //    assertThat(config.targetSystems).containsOnly("mytargetsystem");
-  //  }
-  //
-  //  @Test
-  //  @SetSystemProperty(key = "otel.metric.export.interval", value = "123")
-  //  void otelMetricExportIntervalRespected() {
-  //    JmxConfig config = new JmxConfig();
-  //    assertThat(config.intervalMilliseconds).isEqualTo(10000);
-  //    assertThat(config.properties.getProperty("otel.metric.export.interval")).isEqualTo("123");
-  //  }
-  //
+  @ParameterizedTest
```

Comment:
[for reviewer] we could do the same tests with "real" yaml systems definitions, but this is not possible because we need to cover all the cases where a system is exclusively supported in `legacy` or `instrumentation` and when it's in both, in which case the priority will depend on the value of `otel.jmx.target.source`.